app/controllers/admin/groups_controller.rb

```diff
@@ -0,0 +1,63 @@
+class Admin::GroupsController < ApplicationController
```

Comment:
Sorry, a bit late but why don't you rename the current `admin` controller to `admin/index` and create an `AdminController` class that would encapsulate the authentication logic, `set_admin`, etc. ? 

Maybe in another PR, I don't mind, just asking what you think about this. :-)